adapters/data_parser.py

```diff
@@ -45,8 +47,22 @@ def read_scores(file_path: str) -> list[Student]:
     return results
 
 
-def display_scores(results: list[Student]) -> None:
-    print("*** Created at :", datetime.datetime.now().strftime('%Y-%m-%d %H:%M'))
+def general_summary(results: list[Student], output_path: str, save_as_txt: bool, save_as_pdf: bool,
+                    display_summary: bool, txt_writer: FileWriter, pdf_writer: FileWriter) -> Optional[str]:
+    summary = f'*** Created at: {datetime.datetime.now().strftime("%Y-%m-%d %H:%M")}\n'
     for student in results:
-        student.display_student_info()
-        print('_ _ _')
+        summary += student.display_student_info() + '\n'
+        summary += '_ _ _\n'
+
+    if save_as_txt:
+        txt_writer.write(summary, output_path)
+    if save_as_pdf:
+        pdf_writer.write(summary, output_path)
+    if display_summary:
+        print_to_terminal(summary)
+
+    return summary if (save_as_txt or save_as_pdf) else None
+
+
+def print_to_terminal(summary: Optional[str]) -> None:
+    print(summary)
```

Comment:
This should be moved to `data_saver` and preferably converted to class deriving from `FileWriter`